File: seleniumcode.js
const { Builder, By } = require('selenium-webdriver');
const path = require('path');

(async function testCalculator() {
  // Get the local file URL
  const filePath = 'file://' + path.resolve('calculator.html');

  // Launch browser
  let driver = await new Builder().forBrowser('chrome').build();

  try {
    // Open the calculator page
    await driver.get(filePath);

    // Enter numbers
    await driver.findElement(By.id('num1')).sendKeys('10');
    await driver.findElement(By.id('num2')).sendKeys('25');

    // Click the Add button
    await driver.findElement(By.tagName('button')).click();

    // Wait a bit to allow DOM update
    await driver.sleep(1000);

    // Get result
    const result = await driver.findElement(By.id('result')).getText();
    console.log("Displayed Result:", result);

    // Test condition
    if (result === 'Result: 35') {
      console.log('Test Passed ✅');
    } else {
      console.log('Test Failed ❌');
    }

  } finally {
    await driver.quit();
  }
})();

File: seleniumcode.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time
import os

# Load the local HTML file
file_path = 'file://' + os.path.abspath("Addition_calculator.html")

# Initialize the Chrome driver
driver = webdriver.Chrome()

try:
    # Open the calculator page
    driver.get(file_path)

    # Enter numbers
    driver.find_element(By.ID, "num1").send_keys("10")
    driver.find_element(By.ID, "num2").send_keys("25")

    # Click the Add button
    driver.find_element(By.TAG_NAME, "button").click()

    # Wait a moment to let result update
    time.sleep(1)

    # Get the result text
    result = driver.find_element(By.ID, "result").text
    print("Displayed Result:", result)

    # Optional: Check if the result is correct
    assert result == "Result: 35", "Test Failed!"
    print("Test Passed ")

finally:
    time.sleep(5)
    driver.quit()
